state/application_test.go

```diff
@@ -6016,6 +6016,67 @@ func (s *ApplicationSuite) TestProvisioningState(c *gc.C) {
 	})
 }
 
+func (s *ApplicationSuite) TestUpdateStorageConstraints(c *gc.C) {
```

Comment:
We should add a test for concurrent update of charm URL where the new charm has different storage.
See usages of SetBeforeHooks

Reply:
added the test & thanks ian!